gcputil/credentials.go

```diff
@@ -146,23 +156,46 @@ func PublicKey(pemString string) (interface{}, error) {
 	return cert.PublicKey, nil
 }
 
+// ServiceAccountPublicKey returns the public key with the given key ID for
+// the given service account if it exists. If the key does not exist, an error
+// is returned.
 func ServiceAccountPublicKey(serviceAccount string, keyId string) (interface{}, error) {
-	keyUrl := fmt.Sprintf(serviceAccountPublicKeyUrlTemplate, url.PathEscape(serviceAccount))
-	res, err := cleanhttp.DefaultClient().Get(keyUrl)
+	return ServiceAccountPublicKeyWithEndpoint(context.Background(), serviceAccount, keyId, "")
+}
+
+// ServiceAccountPublicKeyWithEndpoint returns the public key with the given key
+// ID for the given service account if it exists. If endpoint is provided, it will
+// be used as the service endpoint for the request. If endpoint is not provided,
+// a default of "https://www.googleapis.com" will be used. If the key does not exist,
+// an error is returned.
+func ServiceAccountPublicKeyWithEndpoint(ctx context.Context, serviceAccount, keyID, endpoint string) (interface{}, error) {
+	if endpoint == "" {
+		endpoint = defaultGoogleAPIsEndpoint
+	}
+
+	keyURLPath := fmt.Sprintf(serviceAccountPublicKeyURLPathTemplate, url.PathEscape(serviceAccount))
+	keyURL := strings.TrimSuffix(endpoint, "/") + keyURLPath
+	req, err := http.NewRequest(http.MethodGet, keyURL, nil)
 	if err != nil {
 		return nil, err
 	}
-	if err := googleapi.CheckResponse(res); err != nil {
+	resp, err := cleanhttp.DefaultClient().Do(req.WithContext(ctx))
```

Comment:
Nit: Consider using `http.NewRequestWithContext` to pass in the context instead.

Reply:
Done in 88a934c.